src/Hedgehog/Property.fs

```diff
@@ -81,11 +81,30 @@ module Property =
     let bind (k : 'a -> Property<'b>) (m : Property<'a>) : Property<'b> =
         bindGen (toGen << k) (toGen m) |> ofGen
 
+#if FABLE_COMPILER
+    let private printValue (value) : string =
+        sprintf "%A" value
+#else
+    let private (|IsResizeArray|_|) (candidate : obj) : obj list option =
+        let t = candidate.GetType()
+        // have to use TypeInfo due to targeting netstandard 1.6
+        let t = System.Reflection.IntrospectionExtensions.GetTypeInfo(t)
+        let isList = t.IsGenericType && t.GetGenericTypeDefinition() = typedefof<ResizeArray<_>>
+        if isList
+        then Some (candidate :?> System.Collections.IEnumerable |> Seq.cast<obj> |> List.ofSeq)
+        else None
+
+    let private printValue (value) : string =
+        match value with
+        | IsResizeArray list -> sprintf "%A" list
+        | _ -> sprintf "%A" value
```

Comment:
Now that I remembered that we can't cast to `ResizeArray<obj>`, I don't think the active pattern helps.  I think it is misleading for the pattern to be named `IsResizeArray` but then not return a `ResizeArray<_>`.

What do you think of this code?  I think it is simpler.  In addition to this suggestion, you would also need to make a change below, where I recommend `x |> improveRuntimeType |> sprintf "%A"`.

```suggestion
    let private improveRuntimeType (value: obj) : obj =
#if FABLE_COMPILER
        value
#else
        let t = value.GetType()
        // have to use TypeInfo due to targeting netstandard 1.6
        let t = System.Reflection.IntrospectionExtensions.GetTypeInfo(t)
        let isList = t.IsGenericType && t.GetGenericTypeDefinition() = typedefof<ResizeArray<_>>
        if isList
        then value :?> System.Collections.IEnumerable |> Seq.cast<obj> |> List.ofSeq
        else value
```

Reply:
Done, but I used a different name for the function that feels more descriptive